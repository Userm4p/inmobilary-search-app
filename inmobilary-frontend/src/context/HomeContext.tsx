'use client';
import { IHomeContext } from '@/types/Home.types';
import { createContext } from 'react';

const homeContextInitialValues: IHomeContext = {
  isLoading: false,
  error: null,
  properties: [],
  getProperties: async () => {},
  handleInputChange: (e: React.ChangeEvent<HTMLInputElement>) => {
    e.preventDefault();
  },
  searchFilters: {},
};

export const HomeContext = createContext<IHomeContext>(homeContextInitialValues);
