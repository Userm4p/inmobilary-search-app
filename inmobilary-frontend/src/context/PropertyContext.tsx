import { IPropertyContext } from '@/types/Property.types';
import { createContext } from 'react';

const initialState: IPropertyContext = {
  getProperty: async () => {},
  isLoading: false,
  error: null,
  property: null,
};

export const PropertyContext = createContext<IPropertyContext>(initialState);
