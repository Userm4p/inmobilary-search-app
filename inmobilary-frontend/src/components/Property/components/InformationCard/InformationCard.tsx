import { PropertyContext } from '@/context/PropertyContext';
import { formatDate, formatMoney } from '@/lib/utils';
import React, { useContext } from 'react';

export const InformationCard = () => {
  const { property } = useContext(PropertyContext);

  return (
    <div className="flex flex-col gap-4 p-6 justify-start items-start bg-stone-800 shadow-md w-full md:w-[604px] sm:w-[504px] xs:w-[360px] xxs:w-[350px] w-[300px] h-auto rounded-lg">
      <div className="w-full">
        <h1 className="text-3xl font-bold text-white text-nowrap">
          Price: {formatMoney(property?.price || 0)}
        </h1>
        <hr className="my-3 border-gray-200" />
        <p className="text-md text-gray-100 mb-2">{property?.address}</p>
        <div className="flex flex-row gap-4">
          <span className="text-md font-semibold text-gray-200">Owner: {property?.owner.name}</span>
        </div>
      </div>

      {property?.traces && property.traces.length > 0 && (
        <div className="w-full">
          <hr className="my-4 border-gray-200" />
          <h2 className="text-xl font-bold text-white mb-4">Property History</h2>
          <div className="space-y-3">
            {property.traces.map((trace, index) => (
              <div
                key={trace.id}
                className="bg-gradient-to-r from-blue-50 to-indigo-50 border border-blue-200 rounded-lg p-4 hover:shadow-md transition-shadow duration-200"
              >
                <div className="flex justify-between items-start mb-2">
                  <h3 className="font-semibold text-gray-800 text-lg">{trace.name}</h3>
                  <span className="text-sm text-gray-500 bg-gray-100 px-2 py-1 rounded-full">
                    {formatDate(trace.dateScale)}
                  </span>
                </div>
                <div className="grid grid-cols-2 gap-4 text-sm">
                  <div className="flex flex-col">
                    <span className="text-gray-600 font-medium">Value:</span>
                    <span className="text-green-600 font-bold text-lg">
                      {formatMoney(trace.value)}
                    </span>
                  </div>
                  <div className="flex flex-col">
                    <span className="text-gray-600 font-medium">Tax:</span>
                    <span className="text-red-600 font-bold text-lg">{formatMoney(trace.tax)}</span>
                  </div>
                </div>
                {index < property.traces.length - 1 && (
                  <div className="mt-3 pt-3 border-t border-blue-100"></div>
                )}
              </div>
            ))}
          </div>
        </div>
      )}
    </div>
  );
};
