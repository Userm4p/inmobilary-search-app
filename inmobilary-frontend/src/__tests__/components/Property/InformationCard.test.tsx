import { render, screen } from '@testing-library/react';
import { InformationCard } from '@/components/Property/components/InformationCard/InformationCard';
import { PropertyContext } from '@/context/PropertyContext';

const mockProperty = {
  id: '1',
  name: 'Test Property',
  address: '123 Test St',
  price: 100000,
  ownerId: 'owner1',
  owner: {
    id: 'owner1',
    name: 'John Doe',
    address: '456 Owner St',
    photo: 'photo.jpg',
    birthday: '1990-01-01',
  },
  images: ['image1.jpg', 'image2.jpg'],
  traces: [],
};

const mockContextValue = {
  isLoading: false,
  error: null,
  property: mockProperty,
  getProperty: jest.fn(),
};

describe('InformationCard', () => {
  it('should render property information', () => {
    const { container } = render(
      <PropertyContext.Provider value={mockContextValue as any}>
        <InformationCard />
      </PropertyContext.Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
