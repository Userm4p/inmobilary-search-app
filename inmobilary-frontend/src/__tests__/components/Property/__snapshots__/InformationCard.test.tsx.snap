// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`InformationCard should render property information 1`] = `
<div>
  <div
    class="flex flex-col gap-4 p-6 justify-start items-start bg-stone-800 shadow-md w-full md:w-[604px] sm:w-[504px] xs:w-[360px] xxs:w-[350px] w-[300px] h-auto rounded-lg"
  >
    <div
      class="w-full"
    >
      <h1
        class="text-3xl font-bold text-white text-nowrap"
      >
        Price: 
        $100,000
      </h1>
      <hr
        class="my-3 border-gray-200"
      />
      <p
        class="text-md text-gray-100 mb-2"
      >
        123 Test St
      </p>
      <div
        class="flex flex-row gap-4"
      >
        <span
          class="text-md font-semibold text-gray-200"
        >
          Owner: 
          John Doe
        </span>
      </div>
    </div>
  </div>
</div>
`;
